doc/scalar/riscv-crypto-scalar-entropy-source.adoc

```diff
@@ -106,128 +89,73 @@ false positive in the continuous testing procedures. In case of a fatal
 
 [[crypto_scalar_es_state,reftext="Entropy Source State Transition Diagram"]]
 ====
-image::es_state.svg[title="Entropy Source state transition diagram.", align="center"]
+image::es_state.svg[title="Entropy Source state transition diagram.", align="center",scaledwidth=50%]
 Normally the operational state alternates between WAIT
 (no data) and ES16, which means that 16 bits of randomness (seed)
 have been polled. BIST (Built-in Self-Test) only occurs after reset
 or to signal a non-fatal self-test alarm (if reached after WAIT or
 ES16). DEAD is an unrecoverable error state.
-In test mode (when GetNoise is active), WAIT and ES16 states are
-unavailable in PollEntropy.
 ====
 
-The diagram above illustrates operational state (`OPST`) transitions.
-The state is usually either `WAIT` or `ES16`. There are no mandatory
-interrupts.
-
-
-We recommend against busy-loop polling on the entropy source. Even though
-no specific interrupt sequence is specified, it is expected that the `wfi`
-(wait for interrupt) instruction is available on microcontroller targets.
-The polling mechanism can be implemented in a way that allows generic
-drivers to benefit from masked signals that release the program flow
-from a `wfi` instruction without causing an interrupt handler to be invoked.
-
-As a minimum requirement for portable drivers in microcontrollers,
-a `WAIT` or `BIST` from `sentropy` should be followed by a `wfi`
-before another `sentropy` instruction is issued.
-
-Higher-end systems generally perform opportunistic polling as a part of their
-entropy gathering processes.
-
-There is no need to poll after a `DEAD` state.
-
 [[crypto_scalar_es_req]]
 === Entropy Source Requirements
 
 The output `seed` from `sentropy` is not necessarily fully conditioned
 randomness due to hardware and energy limitations of smaller, low-powered
 implementations. However, minimum requirements are defined.
-A caller must not use the output directly but poll twice the amount
-of required entropy, cryptographically condition (hash) it, and use that
-to seed a cryptographic DRBG.
-
-RISC-V requires drivers to implement at least 2-to-1 cryptographic
-post-processing in software with the expectation that the final output
-from this post-processing would have *full entropy* as defined
-in NIST SP 800-90C cite:[BaKeRo:21]. The technical implication is
-that no single 256-bit sequence should have a predicted probability
-larger than 2^-192^ of occurring (min-entropy 192 bits).
+The main requirement is that 2-to-1 cryptographic post-processing
+in 256-bit input blocks will yield 128-bit "full entropy" output blocks.
+Users of `sentropy` may make this conservative assumption but are not
+prohibited from using more than twice the number of seed bits relative
+to the desired resulting entropy.
 
-Depending on the implementation and process instance, the source can be one of
-three things:
+The entropy source can meet one (or more) of three sets of requirements:
```

Comment:
Aren't these requirements what the arch-review folks wanted in the appendix?

In any case, saying the ES "can meet one (or more) of three sets of requirements" doesn't actually convey a requirement I can test, only a possibility.

Reply:
Well, the idea is that entropy itself is not within the scope of architectural tests, just its behavior in relation to the interface (beginning of Appendix). If the interface is implemented correctly, the back-end entropy source module can be changed. In other words, the "entropy source interface" is within scope, but the "entropy source" is not, right? I hope that Section B.1. "A Design and Self-Certification Checklist" clarifies this a bit.